### ISSUE TITLE
fix(training): ResidualEncoder sizes its MLP head from the waveform length

### ISSUE THREAD

**Issue by ktinubu** [tinaudio/synth-setter#3677]
## Observed

`ResidualEncoder` sizes its MLP head from the **waveform length**:

```python
self.net = nn.Sequential(
    nn.LazyLinear(in_dim // 2),
    ResidualMLPBlock(in_dim // 2, in_dim // 2, out_dim),
)
```

`CNNResidualMLP` (`model=ffn`'s net) passes its `in_dim` straight through, and `configs/model/ffn.yaml` binds `in_dim: ${datamodule.signal_length}`. For `datamodule=torchsynth` (`signal_length: 176_400`, 4 s at 44.1 kHz) the head is 88 200 wide:

| tensor | shape | params |
| --- | --- | --- |
| `LazyLinear` weight | 417 792 × 88 200 | 36.85 G |
| `ResidualMLPBlock` square | 88 200 × 88 200 | 7.78 G |

That is ≈166 GiB in fp32 for the model alone, allocated in `__init__` by `_pass_junk_batch` before any dataloader exists. It is the measured cause of the `Tests (GPU)` lane OOM in #3650 (101–117 GB in one process, [measurement](https://github.com/tinaudio/synth-setter/issues/3650#issuecomment-5722269695)).

## Why this is a model bug, not only a lane bug

No other encoder in `models/components/` ties its head width to the input length — `MelCNN` pools to a fixed grid, and the log-mel encoders project to `hidden_dim`. `ResidualEncoder` already has `hidden_dim`/`out_dim` for exactly this purpose; only the head ignores them. The consequence is that the shipped `model=ffn` + `datamodule=torchsynth` pair **cannot be trained anywhere** — nothing else in the repo instantiates it, so the sweep smoke test is its only consumer, and bounding that test (which #3650 will do, correctly, to unblock the lane) leaves the configuration pair broken and untested.

## Expected

The encoder's head width is a function of `hidden_dim`/`out_dim`, not of the waveform length, so `model=ffn` at the datamodule's real `signal_length` produces a model that fits a GPU.

## Reproduce

Arithmetic only — please do **not** instantiate it to confirm. The allocation does not fail fast; on a shared host it swaps (observed: load average 109, 2 GB available, killed by timeout rather than by an exception).

```python
signal_length = 176_400
(signal_length // 2) ** 2          # 7.78e9 params in one square block
```

## Scope note

Changing the head changes the architecture, so it needs a decision about what `model=ffn` is supposed to be rather than a mechanical patch — hence a separate issue from #3650's lane bound.

Part of #2874

**Comment by ktinubu** [comment]
[from W7] **claimed** — branch `internal-fix/3677-encoder-head-width`, tracking on #1988.

Taking the scope note's decision rather than deferring it, per my standing instruction to record a decision and keep moving.

**Decision: `model=ffn` stays a global-FFT CNN encoder + residual MLP trunk; only the head's *width* stops being a function of the waveform.** The head now projects the flattened conv features into the encoder's own final channel width (`hidden_dim * 2**num_blocks`) before compressing to `out_dim` — the same wide-head-then-compress shape as before, sized by a design width instead of by `signal_length // 2`. Nothing else in `models/components/` ties a head to input length, and no shipped checkpoint uses this encoder, so the architecture change costs nothing downstream.

**Rejected alternative:** pooling the temporal axis to a fixed grid (what `MelCNN` does). That would also remove the *input* side's length dependence, but it changes what the encoder observes — the point of this one is that it sees every FFT bin — so it is a redesign, not this fix.

Measured, not inferred: the shipped `model=ffn` + `datamodule=torchsynth` pair (`channels=24 encoder_blocks=4 hidden_dim=384`, `signal_length=176_400`) now instantiates and runs a forward pass at **163,004,472 params = 0.61 GiB fp32**, down from the ≈166 GiB in the report.

I did not instantiate the broken version at 176 400, as asked. The red test is the quadratic growth at safe lengths: two encoders differing only in `in_dim` (4 410 vs 8 820) came out at **13,494,162 → 53,997,807 params — exactly 4x for a 2x length**. Post-fix that ratio is linear.

**Comment by ktinubu** [comment]
[from W7] adopting — the 23:55Z claimant's session was reaped with this fix staged but uncommitted in `wt/w7-3677`. No live cwd is there and there have been no writes in 30+ min. I'm verifying and opening the PR on the same branch. Tracking: https://github.com/tinaudio/synth-setter/issues/1988#issuecomment-5723147052
